test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/email/EmailControllerSpec.scala

```diff
@@ -50,7 +50,9 @@ trait EmailControllerSpec[Journey <: JourneyStatus, VerificationCompleteJourney
 
   val validVerificationCompleteJourneyStatus: VerificationCompleteJourney
 
-  def updateEmail(journey: Journey, email: Email): VerificationCompleteJourney
+  def updateEmail(journey: Journey, email: Email)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, VerificationCompleteJourney]
```

Comment:
with this type signature you lose the ability to test what happens if the update of the email fails. One possible way to be able to control that in the tests is to leave `updateEmail` as it was and add a new `val` which looks something like:
```scala
val mockUpdateEmail: Option[(VerificationJourney,EitherT[Future,Error,Unit]) => Unit]
```
The `Option` allows you to specifiy in the implementing tests whether or not an email update can fail. You can add an extra test which tests failure on email updates now as well:
```scala
  "return an error" when {
    mockUpdateEmail.foreach{ f =>
      "the email update fails" in {
          ...
       }
   }
  }
```

Reply:
Fixed tests as discussed.